### ISSUE TITLE
Program hangs on exit if duplicate tasks were added

### ISSUE THREAD

**Issue by mcejp** [natcap/taskgraph#64]
Adding a duplicate task will repeatedly increment `_added_task_count`, but `len(self._completed_task_names)` will increase only once, making the task executors hang at this check and never exit: https://github.com/natcap/taskgraph/blob/c39e9ebeac25e7a786de67e93f12f2887f4cbbb1/taskgraph/Task.py#L498

(As a workaround, calling `task_graph._terminate()` at the end of the program terminates the executors by force)

**Comment by richpsharp** [comment]
Thanks for this @mcejp, I've seen this in practice but it's hard to reproduce. I couldn't directly reproduce it on a test case, but I think `pytest` aggressively deletes objects that go out of scope so that would indirectly invoke `_terminate` on the TaskGraph object. I've got a PR in for this that'll make it it through soon.

**Comment by mcejp** [comment]
Thanks ❤️
